code/modules/mob/login.dm

```diff
@@ -54,3 +53,7 @@
 		log_played_names(client.ckey,name,real_name)
 
 	log_message("Client [key_name(src)] has taken ownership of mob [src]([src.type])", LOG_OWNERSHIP)
+
+	//If login fails and there is no client or src, we want to bail, because otherwise there will be no logout
+	if(client)

```

Comment:
I left out src, but I'm almost tempted to add it, even if it seems cargo cult

Reply:
I'm pretty sure the semantics are that if something is hard deleted, any sleeping procs with it as `src` are cancelled